code/modules/power/apc.dm

```diff
@@ -237,16 +237,29 @@ var/zapLimiter = 0
 // update the APC icon to show the three base states
 // also add overlays for indicator lights
 /obj/machinery/power/apc/proc/updateicon()
+	ClearAllOverlays(1)
 	if(opened)
-		icon_state = "[ cell ? "apc2" : "apc1" ]" // if opened, show cell if it's inserted
-		ClearAllOverlays(1)
+		icon_state = "apc1"
+
+		if (cell)
+			// if opened, update overlays for cell
+			var/image/I_cell = SafeGetOverlayImage("cell", 'icons/obj/power.dmi', "apc-[cell.icon_state]")
+			UpdateOverlays(I_cell, "cell", 0, 1)
+
 	else if(emagged)
 		icon_state = "apcemag"
-		ClearAllOverlays(1)
 		return
 	else if(wiresexposed)
 		icon_state = "apcwires"
-		ClearAllOverlays(1)
+		var/image/I_wireorange = SafeGetOverlayImage("wireorange", 'icons/obj/power.dmi', "apccut-orange")
+		var/image/I_wiredarkred = SafeGetOverlayImage("wiredarkred", 'icons/obj/power.dmi', "apccut-darkred")
+		var/image/I_wirewhite = SafeGetOverlayImage("wirewhite", 'icons/obj/power.dmi', "apccut-white")
+		var/image/I_wireyellow = SafeGetOverlayImage("wireyellow", 'icons/obj/power.dmi', "apccut-yellow")
+		UpdateOverlays(isWireColorCut(1) ? I_wireorange : null, "wireorange", 0, 1)
+		UpdateOverlays(isWireColorCut(2) ? I_wiredarkred : null, "wiredarkred", 0, 1)
+		UpdateOverlays(isWireColorCut(3) ? I_wirewhite : null, "wirewhite", 0, 1)
+		UpdateOverlays(isWireColorCut(4) ? I_wireyellow : null, "wireyellow", 0, 1)
```

Comment:
what is 1,2,3,4

do you mean:
```
#define APC_WIRE_IDSCAN 1
#define APC_WIRE_MAIN_POWER1 2
#define APC_WIRE_MAIN_POWER2 3
#define APC_WIRE_AI_CONTROL 4
```
?

Reply:
i think so, i wasnt sure if this would update correctly or if it would spoil which cable is which? ill admit im a little out of my depth with the APC wire colors system so any advice would be appreciated - basically the overlays should correspond specifically to the wire *colors* rather than the wire purposes but im not sure what the correct answer is

Reply:
the answer is yes i'm pretty sure, it's not like the colors change function

Reply:
ok i can make this change then

Reply:
resolved in commit 4b51459